src/subscriptions/SubscriptionsScreen.js

```diff
@@ -53,6 +54,7 @@ class SubscriptionsScreen extends React.Component {
         title="Subscriptions"
         searchBarOnChange={this.handleFilterChange}
         searchBar
+        showCancelIcon={filter.length > 0}
```

Comment:
You would want to move that to <Input /> too. The logic should always be 'if input has no text, do not show'